src/Umbraco.Web.BackOffice/Controllers/MemberController.cs

```diff
@@ -147,138 +187,120 @@ public MemberListDisplay GetListNodeDisplay(string listName)
         [OutgoingEditorModelEvent]
         [MemberSaveValidation]
-        public async Task<ActionResult<MemberDisplay>> PostSave(
-            [ModelBinder(typeof(MemberBinder))]
-                MemberSave contentItem)
+        public async Task<ActionResult<MemberDisplay>> PostSave([ModelBinder(typeof(MemberBinder))] MemberSave contentItem)
         {
+            if (contentItem == null)
+            {
+                throw new ArgumentNullException("The member content item was null");
+            }
 
-            //If we've reached here it means:
+            // If we've reached here it means:
             // * Our model has been bound
             // * and validated
             // * any file attachments have been saved to their temporary location for us to use
             // * we have a reference to the DTO object and the persisted object
             // * Permissions are valid
 
-            //map the properties to the persisted entity
+            // map the properties to the persisted entity
             MapPropertyValues(contentItem);
 
             await ValidateMemberDataAsync(contentItem);
 
-            //Unlike content/media - if there are errors for a member, we do NOT proceed to save them, we cannot so return the errors
+            // Unlike content/media - if there are errors for a member, we do NOT proceed to save them, we cannot so return the errors
             if (ModelState.IsValid == false)
             {
-                var forDisplay = _umbracoMapper.Map<MemberDisplay>(contentItem.PersistedContent);
+                MemberDisplay forDisplay = _umbracoMapper.Map<MemberDisplay>(contentItem.PersistedContent);
                 forDisplay.Errors = ModelState.ToErrorDictionary();
                 return new ValidationErrorResult(forDisplay);
             }
 
-            //We're gonna look up the current roles now because the below code can cause
-            // events to be raised and developers could be manually adding roles to members in
-            // their handlers. If we don't look this up now there's a chance we'll just end up
-            // removing the roles they've assigned.
-            var currRoles = _memberService.GetAllRoles(contentItem.PersistedContent.Username);
-            //find the ones to remove and remove them
-            var rolesToRemove = currRoles.Except(contentItem.Groups).ToArray();
-
-            //Depending on the action we need to first do a create or update using the membership provider
+            // Depending on the action we need to first do a create or update using the membership manager
             // this ensures that passwords are formatted correctly and also performs the validation on the provider itself.
+
             switch (contentItem.Action)
             {
                 case ContentSaveAction.Save:
-                    UpdateMemberData(contentItem);
+                    Task<ActionResult<bool>> updateSuccessful = UpdateMemberAsync(contentItem);
```

Comment:
We don't use these responses do we? same with createSuccessful?

Reply:
We will need to ensure `updateSuccessful.Result` is null, otherwise, it should be returned.

Reply:
@bergmania what should I return if .`Result` is not null, when the controller expects `ActionResult<MemberDisplay>? The remaining controller code already checks for model state errors and displays them as needed

Reply:
@emmagarland , The method has `ActionResult<MemberDisplay>` as return type, so you should be able to return the result

```suggestion
                    ActionResult<bool> updateSuccessful = await UpdateMemberAsync(contentItem);
                    if(!(updateSuccessful.Result  is null){
                        return updateSuccessful.Result;
                    }
                    // else we know the `updateSuccessful.Value` should be used.
```

Reply:
Agree - have amended